src/query/service/src/servers/http/middleware.rs

```diff
@@ -95,22 +95,64 @@ fn get_credential(req: &Request, kind: HttpHandlerKind) -> Result<Credential> {
         let msg = &format!("Multiple {} headers detected", AUTHORIZATION);
         return Err(ErrorCode::AuthenticateFailure(msg));
     }
-    let client_ip = match req.remote_addr().0 {
-        Addr::SocketAddr(addr) => Some(addr.ip().to_string()),
-        Addr::Custom(..) => Some("127.0.0.1".to_string()),
-        _ => None,
-    };
+    let conn_ip = get_connection_ip(&req);
     if std_auth_headers.is_empty() {
         if matches!(kind, HttpHandlerKind::Clickhouse) {
-            auth_clickhouse_name_password(req, client_ip)
+            auth_clickhouse_name_password(req, conn_ip)
         } else {
             Err(ErrorCode::AuthenticateFailure(
                 "No authorization header detected",
             ))
         }
     } else {
-        auth_by_header(&std_auth_headers, client_ip)
+        auth_by_header(&std_auth_headers, conn_ip)
+    }
+}
+
+/// this function tries to get the client IP address from the headers, if not found, it will
+/// return the connection's remote address. This IP address is ONLY used for logging purpose,
+/// never use this ip on authentication or network policy.
```

Comment:
```
fn get_client_ip(req: &Request) -> Option<String> {
    let headers = ["X-Real-IP", "X-Forwarded-For", "CF-Connecting-IP"];

    for &header in headers.iter() {
        if let Some(value) = req.headers().get(header) {
            if let Ok(ip_str) = value.to_str() {
                if header == "X-Forwarded-For" {
                    let first_ip = ip_str.split(',').next().unwrap_or("").trim();
                    if !first_ip.is_empty() {
                        return Some(first_ip.to_string());
                    }
                } else {
                    return Some(ip_str.to_string());
                }
            }
        }
    }

    get_connection_ip(req)
}
```

Reply:
thank you for sugestion, i've changed as your advice, PTAL